src/libs/ReportUtils.ts

```diff
@@ -3283,8 +3283,44 @@ function buildOptimisticExpenseReport(chatReportID: string, policyID: string, pa
  * @param paymentType - IOU paymentMethodType. Can be oneOf(Elsewhere, Expensify)
  * @param isSettlingUp - Whether we are settling up an IOU
  */
-function getIOUReportActionMessage(iouReportID: string, type: string, total: number, comment: string, currency: string, paymentType = '', isSettlingUp = false): [Message] {
+function getIOUReportActionMessage(iouReportID: string, type: string, total: number, comment: string, currency: string, paymentType = '', isSettlingUp = false): Message[] {
     const report = getReport(iouReportID);
+
+    if (type === CONST.REPORT.ACTIONS.TYPE.SUBMITTED) {
+        const policy = getPolicy(report?.policyID);
+        const ownerPersonalDetails = getPersonalDetailsForAccountID(policy?.submitsTo ?? 0);
+        let submittedToDisplayName: string;
+        if (ownerPersonalDetails?.accountID === currentUserAccountID) {
+            submittedToDisplayName = 'yourself';
+        } else {
+            submittedToDisplayName = ownerPersonalDetails?.displayName
+                ? `${ownerPersonalDetails.displayName}${ownerPersonalDetails.displayName !== ownerPersonalDetails.login ? ` (${ownerPersonalDetails.login})` : ''}`
+                : 'Hidden';
```

Comment:
Nope. I just thought it would be better to have `submitted to Hidden` rather than `submitted To`. The case could happen if the personal details list is not ready in onyx but I couldn't reproduce the case honestly.

Reply:
I think once user got invited to workspace, they will be ready in Onyx because we will get these information while opening the App.